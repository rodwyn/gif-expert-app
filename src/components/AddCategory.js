import React, { useState } from 'react'

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = event => {
    setInputValue(event.target.value);
  };
  const handleSubmit = event => { event.preventDefault(); };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </ form>
  )
}
