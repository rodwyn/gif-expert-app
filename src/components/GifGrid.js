import React, {useState, useEffect} from 'react'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({category}) => {
  const [images, setImages] = useState([]);

  useEffect(()=>{
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=gorillaz&limit=10&api_key=swbFspRTjwsaFOAtITNuUWGQ0AQGOedH';
    const response = await fetch(url);
    const {data} = await response.json();
    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">  
        {
          images.map(
            img => 
              <GifGridItem key={img.id} {...img} />
          )
        }
      </div>
    </>
  )
}
