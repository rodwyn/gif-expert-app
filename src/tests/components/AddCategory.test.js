import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Testing the <AddCategory /> component.', ()=>{
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(()=>{
    jest.clearAllMocks;
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Should render the component correctly.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should change the input text value.', () => {
    const input = wrapper.find('input');
    input.simulate('change', {
      target: {
        value: 'new category'
      }
    });
  })
  
  test('Should not post information on submit if value is empty.', () => {
    wrapper.find('form').simulate('submit', {preventDefault(){} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Should call set categories function and clean input text value.', () => {
    const input = wrapper.find('input');
    input.simulate('change', {
      target: {
        value: 'new category'
      }
    });
    wrapper.find('form').simulate('submit', {preventDefault(){}});
    input.simulate('change', {
      target: {
        value: ''
      }
    });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  })
  
  
});