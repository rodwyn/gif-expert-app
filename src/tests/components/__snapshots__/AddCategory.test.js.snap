// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <AddCategory /> component. Should render the component correctly. 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
